crates/ruma-state-res/src/event_auth.rs

```diff
@@ -256,13 +255,13 @@ pub fn auth_check<E: Event>(
             fetch_state(&EventType::RoomMember, target_user.as_str()).as_ref(),
             sender,
             sender_member_event.as_ref(),
-            incoming_event.content(),
-            prev_event,
+            &incoming_event,
             current_third_party_invite,
             power_levels_event.as_ref(),
             fetch_state(&EventType::RoomJoinRules, "").as_ref(),
             join_authed_user.as_deref(),
             join_authed_user_membership,
+            room_create_event.unwrap(),
```

Comment:
Why is an `unwrap()` being introduced here?

Reply:
There's an early return which checks that `room_create_event` is present (see https://github.com/ruma/ruma/pull/970/files#diff-ba8664213abc0993d20b981ec89e105cd7f800a672ab11cb1fff3ca56d46bdadL198-L202). 

I always try to work around `unwrap`s, but I could not find a better way in this case.

Reply:
Seems like we could just change

```rust
let room_create_event = fetch_state(&EventType::RoomCreate, "");

// 3. If event does not have m.room.create in auth_events reject
if room_create_event.is_none() {
    warn!("no m.room.create event in auth chain");

    return Ok(false);
}
```

to something like

```rust
// 3. If event does not have m.room.create in auth_events reject
let room_create_event = match fetch_state(&EventType::RoomCreate, "") {
    Some(ev) => ev,
    None => {
        warn!("no m.room.create event in auth chain");
        return Ok(false);
    },
};
```

right?

Reply:
Oh, yes, that will work. I will change it later. Thanks!

Reply:
`unwrap` removed.